### PR TITLE
ci: Updating k8s test suite for NPM to latest

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Updates test suite being used by NPM to latest to reduce overhead of maintaining forks.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: test/capz/npm/npm-e2e.sh [hunk 1]
```diff
@@ -255,7 +255,7 @@ run_npm_conformance () {
         mkdir npm-kubernetes
         cd npm-kubernetes
         # NOTE: if this is not downloaded every run, then probably need to sleep before the VFP tag verification
-        git clone https://github.com/huntergregory/kubernetes.git --depth=1 --branch=quit-on-failure
+        git clone https://github.com/huntergregory/kubernetes.git --depth=1 --branch=quit-on-failure # Change this as well?
```
Comment (huntergregory) [inline test/capz/npm/npm-e2e.sh:258]: we can leave this one as is. This file is more so a reference on how to verify Windows builds on CAPZ clusters

File: .pipelines/npm/npm-cni-integration-test.yaml [hunk 2]
```diff
@@ -120,7 +120,8 @@ jobs:
           --provider=local \
           --ginkgo.focus="$focus" \
           --ginkgo.skip="NetworkPolicyLegacy|SCTP" \
-          --kubeconfig=$HOME/.kube/config
+          --kubeconfig=$HOME/.kube/config \
+          --ginkgo.timeout="2h"
```
Comment (huntergregory) [inline .pipelines/npm/npm-cni-integration-test.yaml:124]: Noticing we're only setting timeout in this file. Just curious what is the default timeout?

**Review by huntergregory — APPROVED** [review]

File: .pipelines/npm/npm-cni-integration-test.yaml (hunk 2 above)
Comment (jpayne3506) [reply .pipelines/npm/npm-cni-integration-test.yaml:124]: 60m or 1h. Had to set this as the update to the repo enforced some auto timeouts within ginkgo.

**Review by huntergregory — APPROVED** [review]

**Pull request merged**